jpsShared/src/org/jetbrains/jps/incremental/scala/model/HydraSettings.java

```diff
@@ -10,5 +10,9 @@
  */
 public interface HydraSettings extends JpsElement {
   boolean isHydraEnabled();
+  String getNumberOfCores();
   String getHydraVersion();
+  String getSourcePartitioner();
+  String getHydraStorePath();
+  String getProjectRoot();
```

Comment:
Can we have these as Path or because of serialization/deserialization we need to keep using String? (wonder what they do in the Scala support)

Reply:
I think that the return type can be Path but I don't see the benefit of it. Maybe it would be better to return String but in the method implementation to use nio Path.

Reply:
The reason is typesafety - a Path is not any String. Also, this way you can avoid writing `Paths.get(hydraSettings.getProjectRoot).toString`, which looks a bit odd.

Reply:
Yes...but the return type of the method should remain String. Only the implementation of the method should use Paths.get(...).toString

Reply:
Ok. Then it looks good now. Go ahead and squash commits and I'll merge.